go/vt/vttablet/tabletserver/tx_engine.go

```diff
@@ -707,12 +723,19 @@ func (te *TxEngine) Release(connID int64) error {
 
 // beginNewDbaConnection gets a new dba connection and starts a transaction in it.
 // This should only be used to redo prepared transactions. All the other writes should use the normal pool.
-func (te *TxEngine) beginNewDbaConnection(ctx context.Context) (*StatefulConnection, error) {
+func (te *TxEngine) beginNewDbaConnection(ctx context.Context, settingsQuery string) (*StatefulConnection, error) {
 	dbConn, err := connpool.NewConn(ctx, te.env.Config().DB.DbaWithDB(), nil, nil, te.env)
 	if err != nil {
 		return nil, err
 	}
 
+	// If we have a settings query that we need to apply, we do that before starting the transaction.
+	if settingsQuery != "" {
+		if _, err = dbConn.ExecOnce(ctx, settingsQuery, 1, false); err != nil {
+			return nil, err
+		}
+	}
+
 	sc := &StatefulConnection{
 		dbConn: &connpool.PooledConn{
 			Conn: dbConn,
```

Comment:
Let's add a test case if this is not done this way what can fail.

Reply:
Added!